src/_pytest/python.py

```diff
@@ -143,9 +144,29 @@ def async_fail(nodeid: str) -> None:
     fail(msg, pytrace=False)
 
 
+def async_default_arg_warn(nodeid: str, function_name, param) -> None:
+    msg = (
+        "Test function '"
+        + function_name
+        + "' has a default argument '"
+        + param.name
+        + "="
+        + str(param.default)
+        + "'.\n"
+    )
+    warnings.simplefilter("always", PytestDefaultArgumentWarning)
+    warnings.warn(PytestDefaultArgumentWarning(msg))
+
+
 @hookimpl(trylast=True)
 def pytest_pyfunc_call(pyfuncitem: Function) -> object | None:
```

Comment:
I'm not entirely convinced this is the right place for this. I wonder if something like `FixtureManager.getfixtureinfo` (where we already take care of reading the function's arguments) would make more sense maybe?